app/components/Home.tsx

```diff
@@ -0,0 +1,368 @@
+          },
+          () => {
+            this.updatePreviewText();
+            this.loadSavedMode();
+          }
+        );
+      },
+      clearUndoHistory: () => {},
+      getElementsBySelector: () => [] as any,
+      generateCustomPreview: (text: string) => {
+        this.updatePreviewText();
+        const preview = document.getElementById(HtmlElementId.Preview);
+        const previewText = this.truncateString(
+          preview.textContent || preview.innerText
+        );
+        return {
+          html: `<div>${previewText}</div>`,
+          plain: previewText,
+        };
+      },
+    });
+
+    this.editorKit = new EditorKit({
+      delegate: delegate,
+      mode: 'plaintext',
+      supportsFilesafe: false,
+    });
+  }
+
+  saveNote = () => {
+    this.editorKit.onEditorValueChanged(this.state.text);
+  };
+
+  handleInputChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const target = event.target;
+    const value = target.value;
+
+    this.setState(
+      {
+        text: value,
+      },
+      () => {
+        this.saveNote();
+        this.updatePreviewText();
+      }
+    );
+  };
+
+  loadSavedMode = () => {
+    try {
+      const savedMode = this.editorKit.internal.componentManager.componentDataValueForKey(
+        ComponentDataKey.Mode
+      );
+      if (savedMode) {
+        this.setModeFromModeType(savedMode);
+      }
+      this.setState({
+        platform: this.editorKit.internal.componentManager.platform,
+      });
+    } catch {}
+  };
+
+  setModeFromModeType = (value: ModeType) => {
+    // Can be refactored to use find
+    // const mode = modes.find // where mode.type = value
+    for (const mode of modes) {
+      if (mode.type === value) {
+        this.logDebugMessage('setModeFromModeType mode: ', mode.type);
+        this.setState(
+          {
+            mode,
+          },
+          () => {
+            this.updatePreviewText();
+          }
+        );
+        return;
+      }
+    }
+  };
+
+  changeMode = (mode: Mode) => {
+    this.setState(
+      {
+        mode,
+      },
+      () => {
+        this.updatePreviewText();
+      }
+    );
+    this.logDebugMessage('changeMode mode: ', mode.type);
+    this.editorKit.internal.componentManager.setComponentDataValueForKey(
+      ComponentDataKey.Mode,
+      mode.type
+    );
+  };
+
+  configureMarkdown = () => {
+    // Remember old renderer, if overriden, or proxy to default renderer
+    const defaultRender =
+      MarkdownParser.renderer.rules.link_open ||
+      function (tokens: any, idx: any, options: any, env: any, self: any) {
+        return self.renderToken(tokens, idx, options);
+      };
+
+    MarkdownParser.renderer.rules.link_open = function (
+      tokens: any,
+      idx: any,
+      options: any,
+      env: any,
+      self: any
+    ) {
+      // If you are sure other plugins can't add `target` - drop check below
+      let aIndex = tokens[idx].attrIndex('target');
```

Comment:
Should be `const`. Also no newline below needed.

Reply:
OK 👍